src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -995,6 +996,9 @@ ReactDOMComponent.Mixin = {
       }
       if (propKey === STYLE) {
         if (nextProp) {
+          if (Array.isArray(nextProp)) {
```

Comment:
I'd love to get feedback from @sebmarkbage on how to integrate with the React reconciliation. This is how we started with React native but now we are first doing a pass to check if something has changed in order to not have to flatten when that's not the case.